### ISSUE TITLE
httplib::Server Put node is only called when the Request contains a body

### ISSUE THREAD

**Issue by schnudi** [yhirose/cpp-httplib#1304]
When adding the Put method a following:

```
svr.Put("/node", [&](const httplib::Request &req, httplib::Response &res) {
    // debug ptint
    for (auto h : req.headers) {
        std::cout << h.first << " " << h.second << std::endl;
    }
    std::cout << req.body << std::endl;

    bool has_param_data = req.has_param("data");
    if(!has_param_data) {
        res.status = 400;
        return;
    }
    /* do something */
    res.set_content("{ "message" : "test" }", "application/json");
});
```
The method is only called when the Request containing a body. If no body is present in the Request, the `set_error_handler` is called.

Is there any option to call the registered Put method also when the body is empty? Or is it maybe a bug in the library?

**Comment by yhirose** [comment]
@schnudi, does HTTP your client set 'Content-Length: 0' to the request headers?

**Comment by schnudi** [comment]
Hi @yhirose 
Thanks for the fast reply! 

I checked my HTTP client (which is a third party executable) in detail. Some how came, that the Client send an OPTIONS request before sending the expected PUT request. In the first step, I just saw that `set_error_handler` was called...

Anyway, handling the OPTIONS request correctly in my Server implementation allows now to run the PUT request as expected.

Sorry for the noise in this thread, and many thanks for your fast reply!
